### DISCUSSION TITLE
Gaps in fill with some icons

### DISCUSSION THREAD

**Discussion by MattRey** [tailwindlabs/heroicons#927] [Help]
Hello! I'm importing a relatively simple icon into a NextJS project using the @heroicons package and am experiencing some odd behaviour in the fill of the svg which I'd love some help with 🤔 

I'm importing icons into my components using `import {CheckCircleIcon} from '@heroicons/react/24/outline'` and then using them in the JSX like this:
```
<CheckCircleIcon
  height={75}
  className="text-primary-600"
  strokeWidth={0.5}
  fill="#ccc"
  viewBox='2.5 2.25 18.75 19.5'
/>
```
Most icons work fine, however some (this one included) have gaps in the fill like this:
<img width="95" alt="Screenshot 2023-02-07 at 17 41 20" src="https://user-images.githubusercontent.com/16228286/217323839-c3c19f2e-1b11-4644-9a24-aba3f3865334.png">

Have tried setting `fillRule`, `clipRule`, `strokeLineCap` etc. but no change.

I'm using `"tailwindcss": "^3.2.4"` and `"@heroicons/react": "^2.0.13"`

Any tips greatly appreciated 🙏

**Comment by JameelKhan9** [comment]
Facing the same issue, and would appreciate some guidance.

**Comment by zefir-git** [comment]
Not sure you are supposed to fill the "outline" icons. Maybe try using the "solid" ones.

If you really want to use outline, put it in a div and give the div background.

**Reply by MattRey** [reply]
I don't really want to have to nest every icon component in a div to adjust background colour. The solid icons are also just solid colour with a stroke icon and not what I need. Because it has been made available on the component, I would expect the fill prop to behave in the same way on each one, or at least guidance for these edge cases to be given 🤷

**Reply by zefir-git** [reply]
> I don't really want to have to nest every icon component in a div to adjust background colour. The solid icons are also just solid colour with a stroke icon and not what I need. Because it has been made available on the component, I would expect the fill prop to behave in the same way on each one, or at least guidance for these edge cases to be given shrug

What you could try to do is separate the check mark and the circle using Inkscape using <kbd>F2</kbd> mode. Afaik by default they all use 1 path so that's probably what causes your issue if you really want to create two-tone icons. Not sure you would want to edit every icon manually though. But in any case with just `fill` I don't think this will work. I don't use react so I have no clue what would be the easiest approach for you to implement this.
